### PR TITLE
Add run and join command line args supporting

### PR DESCRIPTION
**Pull request by Stevent-fei**
### Describe what this PR does / why we need it

1. When the clusterfile is empty, use the mirror to pull up the cluster

### CODE REVIEW

File: cmd/sealer/cmd/run.go [hunk 1]
```diff
@@ -67,13 +72,44 @@ create a cluster with custom environment variables:
...
+		ClusterImageName := args[0]
 		if clusterFile != "" {
 			var err error
 			cf, err = clusterfile.NewClusterFile(clusterFile)
 			if err != nil {
 				return err
 			}
 		}
+		if clusterFile == "" {
+			if err := validateArgs(runArgs); err != nil {
```
Comment (VinceCui) [inline cmd/sealer/cmd/run.go:84]: packaged into a func.

File: cmd/sealer/cmd/hosts.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+	if isRange(inNodes) {
+		if inNodes, err = utilsnet.IPRangeToList(inNodes); err != nil {
+			return nil, err
+		}
+	}
+
+	masters := strings.Split(inMasters, ",")
+	masterHosts := make([]v2.Host, 0, len(masters))
+	for index, master := range masters {
```
Comment (VinceCui) [inline cmd/sealer/cmd/hosts.go:47]: just put all masters into a v2.Host, and no need the MASTER0 role, this role will be automatically assigned without manual

File: cmd/sealer/cmd/hosts.go [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+			IPS: []net.IP{
+				net.ParseIP(master),
+			},
+			Roles: []string{common.MASTER},
+		})
+	}
+
+	// if inNodes is empty,Split will return a slice of length 1 whose only element is inNodes.
+	// so we need to filter the empty string to make sure the cluster node ip is valid.
+	nodes := strings.Split(inNodes, ",")
```
Comment (VinceCui) [inline cmd/sealer/cmd/hosts.go:70]: put all nodes into a v2.Host

**Review by VinceCui — COMMENTED** [review]
i think hosts.go and validate.go should not in cmd package, move them to apply or utils.

File: utils/runcmd/constructClusterFromArg.go [hunk 4]
```diff
@@ -0,0 +1,40 @@
+package runcmd
```
Comment (kakaZhou719) [inline utils/runcmd/constructClusterFromArg.go:1]: These methods are strongly related to run and join cmd, and it is better to put them in cmd pkg.

File: cmd/sealer/cmd/join.go [hunk 5]
```diff
@@ -50,6 +55,42 @@ join default cluster:
...
+					return fmt.Errorf("failed to validate input scale nodes ip: %v", err)
+				}
+			}
+
+			cluster := &v2.Cluster{}
+			if err := yaml.UnmarshalFile(cf, cluster); err != nil {
+				return err
+			}
+
+			if cluster.Name == "" {
```
Comment (kakaZhou719) [inline cmd/sealer/cmd/join.go:90]: why  we need to judge  `cluster.Name`  again  after writing cluster file.

File: cmd/sealer/cmd/join.go [hunk 6]
```diff
@@ -50,6 +55,42 @@ join default cluster:
...
+
+			// validate input nodes IP info
+			if len(joinArgs.Nodes) != 0 {
+				if err := runcmd.ValidateIPStr(joinArgs.Nodes); err != nil {
+					return fmt.Errorf("failed to validate input scale nodes ip: %v", err)
+				}
+			}
+
+			cluster := &v2.Cluster{}
+			if err := yaml.UnmarshalFile(cf, cluster); err != nil {
```
Comment (kakaZhou719) [inline cmd/sealer/cmd/join.go:86]: i think no need to write cluster to disk

File: cmd/sealer/cmd/utils/hosts.go [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+	masterHosts := make([]v2.Host, 0, len(masters))
+	for _, master := range masters {
+		masterHosts = append(masterHosts, v2.Host{
+			IPS: []net.IP{net.ParseIP(master)},
+		})
+	}
+	nodes := strings.Split(inNodes, ",")
+	nodeHosts := make([]v2.Host, 0, len(nodes))
+	for _, node := range nodes {
+		if node != "" {
```
Comment (VinceCui) [inline cmd/sealer/cmd/utils/hosts.go:54]: if node == "" {
			continue
		}
               nodeHosts = append(nodeHosts, v2.Host{
			IPS: []net.IP{net.ParseIP(node)},
	        })

File: cmd/sealer/cmd/utils/validate.go [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+	"net"
+	"strings"
+
+	"github.com/sealerio/sealer/apply"
+
+	netutils "github.com/sealerio/sealer/utils/net"
+)
+
+// ValidateArgs validates all the input args from sealer run command.
+func ValidateArgs(runArgs *apply.Args) error {
```
Comment (VinceCui) [inline cmd/sealer/cmd/utils/validate.go:28]: ValidateRunArgs

File: cmd/sealer/cmd/join.go [hunk 9]
```diff
@@ -43,23 +49,49 @@ join default cluster:
...
+		var cluster *v2.Cluster
 		if clusterFile != "" {
 			var err error
 			cf, err = clusterfile.NewClusterFile(clusterFile)
 			if err != nil {
 				return err
 			}
 		}
+		if clusterFile == "" {
+			var err error
```
Comment (VinceCui) [inline cmd/sealer/cmd/join.go:61]: Like ValidateRunArgs, you can pacakge this block into ValidateJoinArgs

**Review by VinceCui — COMMENTED** [review]
All func in utils, please add unit test.

**Pull request closed without merging**